course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,164 @@
+#include <iostream>
+#include <cstring>
+#include "my_string.h"
+
+MyString::MyString(){}
+
+MyString::MyString(const char* argString)
+{
+        buffer = std::unique_ptr<char[]>(new char[ std::strlen(argString) + 1]);
+        std::strcpy(buffer.get(), argString);;
+}
+
+MyString::MyString(const MyString& argString)
+{
+        *this = argString;
+}
+
+
+MyString::MyString(MyString&& argString){
+        *this = std::move(argString);
+}
+
+const bool MyString::empty() const
+{
+        if (buffer){
+                return buffer[0] == '\0' || buffer[0] == ' ';
+        }
+        else return true;
+}
+
+const size_t MyString::size() const
+{
+        if (buffer){
+                return std::strlen(buffer.get());
+        }
+        else return 0;
+}
+
+const char* MyString::c_str() const
+{
+        if (buffer){
+                return buffer.get();
+        }
+        else {
+            char* r = new char('\0');
```

Comment:
And when is the buffer deleted?
As I said: any explicit "new" or "delete" in your code should be considered a bug.